File: c++_Programs/bubble_sort.cpp
#include <iostream>
#include <vector>

void bubble_sort(std::vector<int>& a) {
    bool swapped;
    for (size_t i = 0; i < a.size(); ++i) {
        swapped = false;
        for (size_t j = 1; j + i < a.size(); ++j) {
            if (a[j-1] > a[j]) {
                std::swap(a[j-1], a[j]);
                swapped = true;
            }
        }
        if (!swapped) break;
    }
}

int main() {
    std::vector<int> v = {5, 2, 9, 1, 5, 6};
    bubble_sort(v);
    for (int x : v) std::cout << x << " ";
    std::cout << std::endl;
    return 0;
}

File: Python_Programs/bubble_sort.py
from typing import List

def bubble_sort(arr: List[int]) -> None:
    """Sort array in-place using bubble sort."""
    n = len(arr)
    for i in range(n - 1):
        swapped = False
        for j in range(n - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break

def main():
    # Example usage
    numbers = [64, 34, 25, 12, 22, 11, 90]
    print(f"Before sorting: {numbers}")
    bubble_sort(numbers)
    print(f"After sorting:  {numbers}")

if __name__ == "__main__":
    main()